gitlab.tf

```diff
@@ -27,3 +27,39 @@ resource "null_resource" "set_gitlab_personal_access_token" {
     null_resource.helmfile_deployments
   ]
 }
+
+resource "null_resource" "add_jenkins_to_gitlab_applications" {
+  triggers = {
+    uuid = uuid()
+  }
+  provisioner "local-exec" {
+    command = "kubectl exec -n gitlab -c task-runner $(kubectl get pod -n gitlab -l \"app=task-runner\" -o jsonpath='{.items[0].metadata.name}') -- gitlab-rails runner 'Doorkeeper::Application.create(uid: \"'\"$JENKINS_ID\"'\", name: \"'\"jenkins\"'\", redirect_uri: \"'\"https://$JENKINS_HOST_NAME/securityRealm/finishLogin\"'\", trusted: true, confidential: true, scopes: [:api, :read_user, :read_api, :read_repository, :write_repository, :read_registry,  :write_registry], secret: \"'\"$JENKINS_SECRET\"'\")'"
+    environment = {
+      KUBECONFIG        = abspath(local_file.kubeconfig.filename)
+      JENKINS_ID        = random_password.jenkins_id.result
+      JENKINS_HOST_NAME = var.jenkins_host_name
+      JENKINS_SECRET    = random_password.jenkins_secret.result
+    }
+  }
+  depends_on = [
+    null_resource.set_gitlab_personal_access_token
+  ]
+}
+
+resource "null_resource" "set_api_token_for_jenkins" {
```

Comment:
Do we really need this? We already generate one for terraform to use that is attached to the root user. Can we just reuse that one?

Reply:
I've successfully deployed using the existing token instead of creating a new one. I've removed the new token creation.